### PR TITLE
Spooky Spectral Instruments

### PR DESCRIPTION
**Pull request by Mercenaryblue**
:cl: 
Add: Skeletons can now have their own spectral instruments
Add: \[Dooting Intensifies\]
admin: the variable "too_spooky" defines if it will spawn new instruments, by default TRUE.
/:cl:

## **Things are about to get Spooky**
Imagine the scariest thing you can.
No, not that old god rising from the sacrifice of the blood cult.
No, not the singularity eating what little safe space you had.
No, not the artifical intelligence deciding to purge all organic lifeforms.

**S K E L E T O N S.**
You have one inside you, right now.
Its waiting the right moment to get out.
To rip apart its fleshy prison.
And play some mournful jazz.
To bring about the spooky-pocalypse.

And the instrument of your doom is a trumpet.
Or a trombone, really.
Even a saxophone could do.
And they'll play it right into your soul.

<a href="https://imgur.com/Raa7fzo"></a>
Happy Spookyween

**What is a spectral trumpet?**
Its an instrument only a skeleton can use. When attacked by it, you receive stamina damage until you give up your mortal life, at which point you will rise as a new skeleton. Otherwise, it acts the same as any other instrument. Also comes with saxophones and trombone variety.

placeholder trumpet sprite: <a href="https://imgur.com/BTb7PMu"></a>
_p.s: trumpets currently use the trombone in-hands and instrument sounds. Just play in high notes._

### CODE REVIEW

**Comment by Mercenaryblue** [comment]
That IconDiffBot is  👍 👍 👍

**Comment by AnonymousNow** [comment]
Isn't the trinity of skeleton instruments the trombone, the xylophone and the bongos? Are we getting those?

File: code/game/objects/items/devices/instruments.dm [hunk 1]
```diff
@@ -110,20 +110,54 @@
...
+
 /obj/item/device/instrument/trombone
 	name = "trombone"
 	desc = "How can any pool table ever hope to compete?"
 	icon_state = "trombone"
 	item_state = "trombone"
 	instrumentId = "trombone"
 
+/obj/item/device/instrument/brass/trombone/spectral
+	name = "spectral trombone"
```
Comment (RemieRichards) [inline code/game/objects/items/devices/instruments.dm]: trom BONE

Comment (Thunder12345) [reply code/game/objects/items/devices/instruments.dm]: 2spooky

File: code/game/objects/items/devices/instruments.dm [hunk 2]
```diff
@@ -164,3 +198,60 @@
...
+		C.stuttering = 20
+		C.adjustStaminaLoss(25) //only humanoids lose the will to live
+		to_chat(C, "<font color='red' size='4'><B>DOOT</B></span>")
+
+		if((H.getStaminaLoss() > 95) && (!istype(H.dna.species, /datum/species/skeleton)) && (!istype(H.dna.species, /datum/species/golem)) && (!istype(H.dna.species, /datum/species/android)) && (!istype(H.dna.species, /datum/species/jelly)))
+			H.Knockdown(30) //doot!
+			H.set_species(/datum/species/skeleton)
+			H.visible_message("<span class='warning'>[H] has given up on life as a mortal.</span>")
+			to_chat(H, "<B>You are the spooky skeleton!</B>")
+			to_chat(H, "A new life and identity has begun. Help your fellow skeletons into bringing out the spooky-pocalypse. You haven't forgotten your past life, but you are no longer beholden to any past loyalties.")
```
Comment (DoctoreMad) [inline code/game/objects/items/devices/instruments.dm]: I believe you need admin ruling to be able to say "Past allegiances are foregone" in this text unless there's something I don't know going about.

File: code/game/objects/items/devices/instruments.dm [hunk 3]
```diff
@@ -164,3 +198,60 @@
...
+			to_chat(U, "<span class= 'danger'> Your ears weren't meant for this spectral sound.</span>")
+			return ..()
+
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(istype(H.dna.species, /datum/species/skeleton))
+			return ..() //undead are unaffected by the spook-pocalypse.
+		if(istype(H.dna.species, /datum/species/zombie))
+			H.adjustStaminaLoss(25) //zombies can't resist the doot
+		C.Knockdown(15)
```
Comment (kevinz000) [inline code/game/objects/items/devices/instruments.dm]: One hit stunlock convert that isn't counter able is shit.

File: code/game/objects/items/devices/instruments.dm (hunk 2 above)
Comment (Mercenaryblue) [reply code/game/objects/items/devices/instruments.dm]: They can only be spawned by admins anyway.

File: code/game/objects/items/devices/instruments.dm (hunk 3 above)
Comment (Mercenaryblue) [reply code/game/objects/items/devices/instruments.dm]: I could alter it to be simply raw stamina loss.

Comment (Mercenaryblue) [reply code/game/objects/items/devices/instruments.dm]: Done. They will no longer stun living creatures.

File: code/game/objects/items/devices/instruments.dm [hunk 4]
```diff
@@ -164,3 +198,58 @@
...
+
+	else
+		C.Jitter(15)
+		C.stuttering = 20
+
+/obj/item/device/instrument/brass/proc/change_name(mob/living/carbon/human/H)
+	var/t = stripped_input(H, "Enter your new skeleton name", H.real_name, null, MAX_NAME_LEN)
+	if(!t)
+		t = "spooky skeleton"
+	H.real_name = t
```
Comment (ShizCalev) [inline code/game/objects/items/devices/instruments.dm]: H.fully_replace_character_name(H.real_name, t)

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by DoctoreMad** [comment]
Make double-haunted version with the same procs except when a person is converted they also get a double-haunted instrument.

File: code/game/objects/items/devices/instruments.dm [hunk 5]
```diff
@@ -164,3 +202,68 @@
...
+				if(prob(30))
+					new/obj/item/device/instrument/brass/saxophone/spectral(T)
+				else if(prob(30))
+					new/obj/item/device/instrument/brass/trumpet/spectral(T)
+				else if(prob(30))
+					new/obj/item/device/instrument/brass/trombone/spectral(T)
+				else
+					to_chat(H, "The spooky gods forgot to ship your instrument. Better luck next unlife.")
+			to_chat(H, "<B>You are the spooky skeleton!</B>")
+			to_chat(H, "A new life and identity has begun. Help your fellow skeletons into bringing out the spooky-pocalypse. You haven't forgotten your past life, but you are no longer beholden to any past loyalties.")
```
Comment (DoctoreMad) [inline code/game/objects/items/devices/instruments.dm]: What happened? You reverted it to the old text. I assume a admin/coder told you that it didn't matter?

Comment (Mercenaryblue) [reply code/game/objects/items/devices/instruments.dm]: Wow. I was certain I changed that. I guess I may have reverted it by accident, or forgot to copy-paste that part over.

**Comment by DoctoreMad** [comment]
For reference, it reverted on your last git push, "Requested changes"

**Comment by Mercenaryblue** [comment]
It was probably my sticky middle mouse at fault.
The requested changes are all there. The PR is ready.

**Comment by DoctoreMad** [comment]
ShizCalev's requested change still shows up as being requested even though the code change is done
What gives

**Comment by Mercenaryblue** [comment]
I think he has to change it manually.

File: code/game/objects/items/devices/instruments.dm [hunk 6]
```diff
@@ -110,20 +111,57 @@
...
 	instrumentId = "saxophone"
 
+/obj/item/device/instrument/brass/saxophone/spectral
+	name = "spectral saxophone"
+	desc = "This spooky sound will be sure to leave mortals in bones."
+	icon_state = "saxophone"
+	item_state = "saxophone"
+	instrumentId = "saxophone"
+	force = 0
+	attack_verb = list("played","jazzed","saxxed","mourned","dooted")
```
Comment (AnturK) [inline code/game/objects/items/devices/instruments.dm]: Dunno if it's intended but you didn't set the spooky here

File: code/game/objects/items/devices/instruments.dm [hunk 7]
```diff
@@ -9,6 +9,7 @@
 	var/datum/song/handheld/song
 	var/instrumentId = "generic"
 	var/instrumentExt = "mid"
+	var/too_spooky = FALSE //this define if the instrument will spawn a copy upon dooting a new skeleton.
```
Comment (AnturK) [inline code/game/objects/items/devices/instruments.dm]: This one is bit ugly. Do instruments even have any unique properties besides sound id and visuals so it can't be spectral/trumpet, spectral/saxophone etc ?

Comment (Mercenaryblue) [reply code/game/objects/items/devices/instruments.dm]: They all behave the same, the only difference is which sound board they use.

File: code/game/objects/items/devices/instruments.dm [hunk 8]
```diff
@@ -143,7 +143,7 @@
 	instrumentId = "saxophone"
 	force = 0
 	attack_verb = list("played","jazzed","saxxed","mourned","dooted")
-
+	too_spooky = TRUE
```
Comment (DoctoreMad) [inline code/game/objects/items/devices/instruments.dm]: Why did you add this here if you didn't remove the flag in the base /instrument item

Comment (Mercenaryblue) [reply code/game/objects/items/devices/instruments.dm]: To allow it to spawn new instruments.

File: code/game/objects/items/devices/instruments.dm (hunk 7 above)
Comment (AnturK) [reply code/game/objects/items/devices/instruments.dm]: Could you make the spectralness a component please ? It just needs to do things on on hit after all, and wouldn't need any weird pathing or special variables  on base types that way. (And could be applied to anything for admin abuse)

Comment (Mercenaryblue) [reply code/game/objects/items/devices/instruments.dm]: I'll do my best.

**Comment by DoctoreMad** [comment]
@ShizCalev Confirm this change reeeee
